### PR TITLE
Enhance decorator unpack_request_args to accept normal functions

### PR DESCRIPTION
**Pull request by semigodking**

### CODE REVIEW

**Comment by fscherf** [comment]
@webknjaz 
@ZhukovGreen 

any opinions?

**Comment by webknjaz** [comment]
I'm not sure why one would want to use blocking functions in an asynchronous env. I believe that interface should be standardized. For CPU-bound tasks one would probably be forced to go asynchronous and spawn a thread anyway, so that a long-running computation-expensive task would get properly scheduled for execution in a non-blocking manner.

Diverse interfaces are misleading. I'd lock that to async-only.

**Comment by semigodking** [comment]
For some trivial functions like short string manipulations, I would prefer
making them normal functions for two considerations:
1. Code could be easily reused by non-aio aware code.
2. Going through event loop is over kill for such functions due to
overhead.

**Comment by webknjaz** [comment]
@semigodking why would you concatenate two strings **on the other machine _over the network_**. Isn't this overhead?

Anyway, exposing such things to non-experienced users is dangerous as they might go for something time-counsuming and then wonder why the i/o loop is blocked.

P.S. You said that you don't wan't to go through event loop, but you're already in it. You probably meant that you don't want a context switch to happen, but it won't happen inside the handler unless you explicitly yield the control to event loop using `await`.
From framework's point of view calling blocking handler is not a good thing to do at all, I'd be +1 for removing such possibility. In I/O intensive app it much be controlled via event loop.

**Comment by semigodking** [comment]
Hi @webknjaz, the reason that I need something like string manipulations over the network is that I need server to determine how to manipulate strings according to configurations and business logics. Of course, it is not limited to strings. 
About event loop and context switch, agree with you.

**Comment by webknjaz** [comment]
Just to summarize, you can still do:
```python
    async def meth(...):
        return "1" + "3"
```
Without context switching inside of the handler. But trying to force framework to run synchronous handlers you dig into the framework's territory.
aiohttp has dropped support for non-coroutine handlers a while back (deprecated in 2.3.0 on 2017-10-18).

**Comment by fscherf** [comment]
@semigodking 
I agree fully with @webknjaz. Allow users to add sync rpc methods leads to bad designs and conflicts with the base concept of this project.

File: aiohttp_json_rpc/rpc.py [hunk 1]
```diff
@@ -294,12 +294,12 @@ def filter(self, topics):
             raise ValueError
 
         self.state[topic] = data
-
+        msg = None
         for client in self.filter(topic):
             try:
-                await self._ws_send_str(
```
Comment (webknjaz) [inline aiohttp_json_rpc/rpc.py]: These changes are totally unrelated to the topic of this PR

**Comment by fscherf** [comment]
@semigodking i am sorry but i will close this PR. I will not merge code to enable sync code in an async api. And even if i wanted to, your patch only enables it when using `unpack_request_args` and not in `add_methods` which would result in an inconsistent api.

If you have fixes for the pub-sub system i look forward to a new PR but with proper patch description.

**Comment by semigodking** [comment]
Sorry, I was only intent to submit a PR for one commit. I will create a new one.

**Comment by webknjaz** [comment]
@semigodking I suggest you use branches to avoid committing unrelated code in the same PR.
